purchases/src/main/kotlin/com/revenuecat/purchases/google/BillingWrapper.kt

```diff
@@ -743,12 +745,17 @@ internal class BillingWrapper(
      * This prevents ANRs, see https://github.com/android/play-billing-samples/issues/310
      */
     private fun retryBillingServiceConnectionWithExponentialBackoff() {
-        log(LogIntent.DEBUG, BillingStrings.BILLING_CLIENT_RETRY.format(reconnectMilliseconds))
-        startConnectionOnMainThread(reconnectMilliseconds)
-        reconnectMilliseconds = min(
-            reconnectMilliseconds * 2,
-            RECONNECT_TIMER_MAX_TIME_MILLISECONDS,
-        )
+        if (reconnectionAlreadyScheduled) {
+            log(LogIntent.DEBUG, BillingStrings.BILLING_CLIENT_RETRY_ALREADY_SCHEDULED)
+        } else {
+            log(LogIntent.DEBUG, BillingStrings.BILLING_CLIENT_RETRY.format(reconnectMilliseconds))
+            reconnectionAlreadyScheduled = true
+            startConnectionOnMainThread(reconnectMilliseconds)
+            reconnectMilliseconds = min(
+                reconnectMilliseconds * 2,
+                RECONNECT_TIMER_MAX_TIME_MILLISECONDS,
+            )
+        }
```

Comment:
I am wondering if we should change these logs from debug to warning? That way they would show up in crash reporting tools logs?

Reply:
Hmm, I guess it depends on how it's configured but yeah, considering it shouldn't happen often, I think that's a good idea.

Reply:
Changed in https://github.com/RevenueCat/purchases-android/pull/1300/commits/4bb42b6acf9400295b2dfe0ee25b1f83418db2dc